Core/Layer/Worlds/StatusBar/StatusBarRenderer.cs

```diff
@@ -0,0 +1,1160 @@
+
+        Color? drawColor = null;
+        if (!string.IsNullOrEmpty(translation))
+        {
+            if (StandardTextColors.TryGetValue(translation, out var c))
+                drawColor = c;
+        }
+
+        if (hud.Textures.HasImage(patch))
+        {
+            hud.Image(patch, drawPos, anchor: align, alpha: alpha, color: drawColor);
+        }
+        else if (hud.Textures.HasImage(patch, ResourceNamespace.Sprites))
+        {
+            hud.Image(patch, drawPos, anchor: align, resourceNamespace: ResourceNamespace.Sprites, alpha: alpha, color: drawColor);
+        }
+    }
+
+    private void DrawNumber(IHudRenderContext hud, StatusBarNumberDef number, Vec2I parentPos, int containerHeight, 
+        StatusBarContext context, bool isPercent, float widescreenOffset)
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, number.Conditions))
+            return;
+
+        int value = ResolveNumberValue(context.Player, number.Type, number.Param);
+        
+        if (number.MaxLength > 0)
+        {
+            int maxVal = (int)Math.Pow(10, number.MaxLength) - 1;
+            int minVal = -(int)Math.Pow(10, number.MaxLength - 1) + 1;
+            if (value > maxVal) value = maxVal;
+            if (value < minVal) value = minVal;
+        }
+
+        var sbarDef = WorldStatic.World.ArchiveCollection.Definitions.StatusBarDefinition;
+        var fontDef = sbarDef.NumberFonts.FirstOrDefault(f => f.Name.EqualsIgnoreCase(number.Font));
+        
+        if (fontDef == null) return;
+
+        string text = value.ToString(CultureInfo.InvariantCulture);
+        if (isPercent) text += "%";
+
+        Vec2I pos = ResolvePosition(number, parentPos, widescreenOffset);
+
+        Color? drawColor = null;
+        if (!string.IsNullOrEmpty(number.Translation))
+        {
+            if (StandardTextColors.TryGetValue(number.Translation, out var c))
+                drawColor = c;
+        }
+
+        float alpha = number.Translucency ? 0.5f : 1.0f;
+
+        int totalWidth = 0;
+        int monoWidth = 0; 
+
+        if (fontDef.Type == 0)
+        {
+            string zeroPatch = GetFontPatch(hud, fontDef, '0');
+            if (hud.Textures.TryGet(zeroPatch, out var zeroHandle))
+                monoWidth = zeroHandle.Dimension.Width;
+        }
+        else if (fontDef.Type == 1)
+        {
+            if (!Type1WidthCache.TryGetValue(fontDef.Stem, out monoWidth))
+            {
+                monoWidth = 0;
+                for (char d = '0'; d <= '9'; d++)
+                {
+                    string dPatch = GetFontPatch(hud, fontDef, d);
+                    if (hud.Textures.TryGet(dPatch, out var dHandle))
+                        monoWidth = Math.Max(monoWidth, dHandle.Dimension.Width);
+                }
+                Type1WidthCache[fontDef.Stem] = monoWidth;
+            }
+        }
+
+        List<(string Patch, int Width, int Offset)> glyphs = new(text.Length);
```

Comment:
Let's make this list a member variable so it can be reused and clear it here. This avoids allocating a new list each time that will get thrown to the GC.

More of a nitpick but I like to use record structs for these:
readonly record struct RenderGlyph(string Patch, int Width, int Offset)

This would give you the explicit property names.